### PR TITLE
TPI-1873: Add steps to help with custom authentication method

### PR DESCRIPTION
**Pull request by rui-lomba**
Jira Ticket: [TPI-1873](https://hyperscience.atlassian.net/browse/TPI-1873)

## Description & Context, Tradeoffs, Considerations, Screenshots
Describe steps to use Custom Authentication Method

image

### CODE REVIEW

**Review by asbestoso — APPROVED** [review]

**Review by Toad2186 — APPROVED** [review]

File: technical-reference.adoc [hunk 1]
```diff
@@ -150,6 +150,19 @@ Uses a token to authenticate the connection.
 
 Uses a custom internal authentication mechanism to connect.
 
+This is the default authentication method for SaaS instances. 
+
+In order to properly authenticate using this method follow these steps:
```
Comment (NatAtHypersci) [inline technical-reference.adoc]:
```suggestion
In order to properly authenticate using this method, follow these steps:
```

File: technical-reference.adoc [hunk 2]
```diff
@@ -150,6 +150,19 @@ Uses a token to authenticate the connection.
...
+
+. Create an API account at Users > API Accounts.
```
Comment (NatAtHypersci) [inline technical-reference.adoc]:
```suggestion
. In the application, create an API account at **Users** > **API Accounts**.
```

File: technical-reference.adoc [hunk 3]
```diff
@@ -150,6 +150,19 @@ Uses a token to authenticate the connection.
 
 Uses a custom internal authentication mechanism to connect.
 
+This is the default authentication method for SaaS instances.
```
Comment (NatAtHypersci) [inline technical-reference.adoc]:
```suggestion
This internal mechanism is the default authentication method for SaaS instances. 
```

File: technical-reference.adoc [hunk 4]
```diff
@@ -150,6 +150,19 @@ Uses a token to authenticate the connection.
 
 Uses a custom internal authentication mechanism to connect.
 
+This is the default authentication method for SaaS instances. 
+
+In order to properly authenticate using this method follow these steps:
+
+. Create an API account at Users > API Accounts.
+. Copy the client ID and account password. You will not be able to access the account password after account creation.
+. Use your client ID in the Client ID field.
```
Comment (NatAtHypersci) [inline technical-reference.adoc]:
```suggestion
. Enter your client ID in the **Client ID** field.
```

File: technical-reference.adoc [hunk 5]
```diff
@@ -150,6 +150,19 @@ Uses a token to authenticate the connection.
...
+. Use your account password in the Client Secret field
```
Comment (NatAtHypersci) [inline technical-reference.adoc]:
```suggestion
. Enter your account password in the **Client Secret** field.
```

File: technical-reference.adoc [hunk 6]
```diff
@@ -150,6 +150,19 @@ Uses a token to authenticate the connection.
...
+. For Auth Server, use the domain of the authentication server, i.e., login.hyperscience.net.
```
Comment (NatAtHypersci) [inline technical-reference.adoc]:
```suggestion
. In the **Auth Server** field, enter the domain of the authentication server (`login.hyperscience.net`).
```

File: technical-reference.adoc [hunk 7]
```diff
@@ -150,6 +150,19 @@ Uses a token to authenticate the connection.
...
+. For Domain, use the domain of the instance where HS app is hosted.
```
Comment (NatAtHypersci) [inline technical-reference.adoc]:
```suggestion
. In the **Domain** field, enter the domain of the instance where the Hyperscience application is hosted (e.g., `example.hyperscience.net`).
```

File: technical-reference.adoc (hunk 6 above)
Comment (NatAtHypersci) [reply technical-reference.adoc]: I'm assuming they would only ever enter `login.hyperscience.net` here? Let me know if I'm wrong about that.

**Review by NatAtHypersci — COMMENTED** [review]
Made a few minor suggestions for clarity / consistency. Please let me know if you have any questions / concerns!

File: technical-reference.adoc (hunk 6 above)
Comment (rui-lomba) [reply technical-reference.adoc]: @NatAtHypersci confirmed with @v1ctorrhs, "depending on the region a different auth server can be used, for example for testing we use login-preview.hyperscience.net, for emea we use login.emea.hyperscience.net"

Comment (NatAtHypersci) [reply technical-reference.adoc]: Ah OK — thanks for clarifying!

Let's use this instead, then: 

. In the **Auth Server** field, enter the domain of the authentication server for your instance (e.g., `login.hyperscience.net`).

**Review by NatAtHypersci — COMMENTED** [review]
Made an additional edit based on our convo in Slack — let me know if you have any questions / concerns about what I wrote!

File: technical-reference.adoc [hunk 8]
```diff
@@ -150,6 +150,19 @@ Uses a token to authenticate the connection.
...
 
+This internal mechanism is the default authentication method for SaaS instances. 
+
+In order to properly authenticate using this method, follow these steps:
+
+. In the application, create an API account at **Users** > **API Accounts**.
+. Copy the client ID and account password. You will not be able to access the account password after account creation.
+. Enter your client ID in the **Client ID** field.
+. Enter your account password in the **Client Secret** field.
+. In the Auth Server field, enter the domain of the authentication server for your instance (e.g., login.hyperscience.net).
```
Comment (NatAtHypersci) [inline technical-reference.adoc]:
```suggestion
. In the **Auth Server** field, enter the domain of the authentication server for your instance (e.g., `login.hyperscience.net`).
```

File: technical-reference.adoc (hunk 6 above)
Comment (rui-lomba) [reply technical-reference.adoc]: Done

**Review by NatAtHypersci — COMMENTED** [review]
Looks like the formatting got lost when I submitted my suggestion — sorry about that!

**Review by NatAtHypersci — APPROVED** [review]
LGTM — thanks for all your help!

**Pull request merged**